diff --git a/src/examples/libpmemobj/tree_map/rbtree_map.c b/src/examples/libpmemobj/tree_map/rbtree_map.c
index 48a4fe671..bfc4d391d 100644
--- a/src/examples/libpmemobj/tree_map/rbtree_map.c
+++ b/src/examples/libpmemobj/tree_map/rbtree_map.c
@@ -111,6 +111,9 @@ rbtree_map_create(PMEMobjpool *pop, TOID(struct rbtree_map) *map, void *arg)
 		D_RW(s)->slots[RB_LEFT] = s;
 		D_RW(s)->slots[RB_RIGHT] = s;
 
+		// BUG: Additional pmemobj_tx_add_range_direct
+		pmemobj_tx_add_range_direct(map, sizeof(*map));
+
 		TOID(struct tree_map_node) r = TX_ZNEW(struct tree_map_node);
 		D_RW(r)->color = COLOR_BLACK;
 		D_RW(r)->parent = s;
